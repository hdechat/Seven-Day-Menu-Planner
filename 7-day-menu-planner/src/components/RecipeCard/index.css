article {
  box-shadow: 0.1rem 0.2rem 0.7rem -0.1rem;
  border: solid gray thin;
  border-radius: 3%;
  margin: 0.5rem auto;
  padding: 0.5rem 0rem;
  width: 100%;
  background: #D5C1B480;
}

img {
  display: block;
  left: 25%;
  margin-bottom: 1rem;
  position: relative;
  width: 50%;
}

a {
  font-size: 12px;
  text-decoration: none;
  color: black;
}
