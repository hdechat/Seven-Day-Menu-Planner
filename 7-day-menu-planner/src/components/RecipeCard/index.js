import React from 'react';
import AddToMenu from '../AddToMenu'

const RecipeCard = ({ recipe }) => {
  return (
    <article>
      <h1>{recipe.title}</h1>
      <img src={recipe.image} alt="recipe meal" />
      <a href={recipe.url}>Click Here to Go To Recipe Page</a>
      <AddToMenu recipe={recipe} />
    </article>
  )
}

export default RecipeCard;