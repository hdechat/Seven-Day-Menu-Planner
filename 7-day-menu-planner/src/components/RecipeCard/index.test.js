import React from 'react';
import { shallow, render } from 'enzyme';
import RecipeCard from './index.js';

describe('Recipe Card', () => {
  let wrapper;
  let mockRecipe;

  beforeEach(() => {
    mockRecipe = {
      "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
      "ingredients": ["1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped", "8 garlic cloves, peeled and smashed", "⅓ cup apple cider vinegar", "3 tablespoons tomato paste", "3 tablespoons honey", "2 tablespoons Paleo-friendly fish sauce (Red Boat!)", "1½ teaspoons kosher salt"],
      "title": "Paleo Sriracha recipes",
      "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha"
    }
    wrapper = shallow(<RecipeCard recipe={mockRecipe} displayAddToMenu={true}/>);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});