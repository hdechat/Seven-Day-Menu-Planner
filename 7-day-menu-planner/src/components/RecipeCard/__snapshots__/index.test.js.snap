// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Card matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeCard
    recipe={
        Object {
            "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
            "ingredients": Array [
              "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped",
              "8 garlic cloves, peeled and smashed",
              "⅓ cup apple cider vinegar",
              "3 tablespoons tomato paste",
              "3 tablespoons honey",
              "2 tablespoons Paleo-friendly fish sauce (Red Boat!)",
              "1½ teaspoons kosher salt",
            ],
            "title": "Paleo Sriracha recipes",
            "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Paleo Sriracha recipes
</h1>,
        <img
          alt="recipe meal"
          src="https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476"
/>,
        <a
          href="http://nomnompaleo.com/post/36060636540/paleo-sriracha"
>
          Click Here to Go To Recipe Page
</a>,
        <Unknown
          recipe={
                    Object {
                              "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
                              "ingredients": Array [
                                "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped",
                                "8 garlic cloves, peeled and smashed",
                                "⅓ cup apple cider vinegar",
                                "3 tablespoons tomato paste",
                                "3 tablespoons honey",
                                "2 tablespoons Paleo-friendly fish sauce (Red Boat!)",
                                "1½ teaspoons kosher salt",
                              ],
                              "title": "Paleo Sriracha recipes",
                              "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Paleo Sriracha recipes",
        },
        "ref": null,
        "rendered": "Paleo Sriracha recipes",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "recipe meal",
          "src": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click Here to Go To Recipe Page",
          "href": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
        },
        "ref": null,
        "rendered": "Click Here to Go To Recipe Page",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "recipe": Object {
            "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
            "ingredients": Array [
              "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped",
              "8 garlic cloves, peeled and smashed",
              "⅓ cup apple cider vinegar",
              "3 tablespoons tomato paste",
              "3 tablespoons honey",
              "2 tablespoons Paleo-friendly fish sauce (Red Boat!)",
              "1½ teaspoons kosher salt",
            ],
            "title": "Paleo Sriracha recipes",
            "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
          },
        },
        "ref": null,
        "rendered": null,
        "type": Object {},
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Paleo Sriracha recipes
</h1>,
          <img
            alt="recipe meal"
            src="https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476"
/>,
          <a
            href="http://nomnompaleo.com/post/36060636540/paleo-sriracha"
>
            Click Here to Go To Recipe Page
</a>,
          <Unknown
            recipe={
                        Object {
                                    "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
                                    "ingredients": Array [
                                      "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped",
                                      "8 garlic cloves, peeled and smashed",
                                      "⅓ cup apple cider vinegar",
                                      "3 tablespoons tomato paste",
                                      "3 tablespoons honey",
                                      "2 tablespoons Paleo-friendly fish sauce (Red Boat!)",
                                      "1½ teaspoons kosher salt",
                                    ],
                                    "title": "Paleo Sriracha recipes",
                                    "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Paleo Sriracha recipes",
          },
          "ref": null,
          "rendered": "Paleo Sriracha recipes",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "recipe meal",
            "src": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click Here to Go To Recipe Page",
            "href": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
          },
          "ref": null,
          "rendered": "Click Here to Go To Recipe Page",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "recipe": Object {
              "image": "https://www.edamam.com/web-img/e50/e50584043c015a91a26a7a37f093d476",
              "ingredients": Array [
                "1½ pounds fresh red jalapeño peppers, stemmed, seeded, and roughly chopped",
                "8 garlic cloves, peeled and smashed",
                "⅓ cup apple cider vinegar",
                "3 tablespoons tomato paste",
                "3 tablespoons honey",
                "2 tablespoons Paleo-friendly fish sauce (Red Boat!)",
                "1½ teaspoons kosher salt",
              ],
              "title": "Paleo Sriracha recipes",
              "url": "http://nomnompaleo.com/post/36060636540/paleo-sriracha",
            },
          },
          "ref": null,
          "rendered": null,
          "type": Object {},
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
